### ISSUE TITLE
[NEXT-472] appdir causes `pages/page.tsx` throws 404 in dev and 500 in prod

### ISSUE THREAD

**Issue by JanKaifer** [vercel/next.js#45651]
reproduction: [https://github.com/JanKaifer/next-repro-45651.git](https://github.com/JanKaifer/next-repro-45651.git)

visit: [http://localhost:3000/page](http://localhost:3000/page)

* [ ] It will throw 404 in dev
* [X]  It will throw 500 in production when loading

To reproduce it in `dev` first navigate to `/` load it and then navigate to `/page`.
(navigating to `/page` first and then to `/` also reproduces the issue)

<sub>From [SyncLinear.com](https://synclinear.com) | [NEXT-472](https://linear.app/vercel/issue/NEXT-472/appdir-causes-pagespagetsx-throws-404-in-dev-and-500-in-prod)</sub>

**Comment by TheHellTower** [comment]
I can't reproduce it.

- `npm -i`
- `npm run dev` / `npm run build & npm run start`

Did I miss something ?

<details>
  <summary> Medias </summary>

![](https://i.imgur.com/GfF4mi8.png)
![](https://i.imgur.com/CIRxUso.gif)

</details>

**Comment by JanKaifer** [comment]
@TheHellTower  check http://localhost:3000/page

**Comment by Rishab49** [comment]
If you want to create a router in next canary then you have to create a folder with the name of route inside the app dir and then create Page.tsx inside it to make it work.
reference : https://beta.nextjs.org/docs/routing/defining-routes#creating-routes

**Comment by gricn** [comment]
@Rishab49 In the reference you provided, it mentions that 

> The app directory works alongside the pages directory to allow for incremental adoption. This allows you to opt some routes of your application into the new behavior while keeping other routes in the pages directory for previous behavior.

**Comment by Rishab49** [comment]
@gricn  Code in the provided repo(https://github.com/JanKaifer/next-repro-45651.git) has `appDir : true` which will try to resolve every request from app directory and not from pages directory. If you want it to handled through pages directory then remove `appDir : true` from the next.config.js file, then the app will start working as intended.

**Comment by gricn** [comment]
> @gricn Code in the provided repo(https://github.com/JanKaifer/next-repro-45651.git) has `appDir : true` which will try to resolve every request from app directory and not from pages directory. If you want it to handled through pages directory then remove `appDir : true` from the next.config.js file, then the app will start working as intended.

It will resolve the files from `app` and `pages` directories with `appDir : true` set. You can read the documentation again or test by adding another folder in the provided repo by yourself.

i.e. 
<img width="313" alt="image" src="https://user-images.githubusercontent.com/34237589/218953279-5f616b57-75e6-4d61-8da4-f9dbe206c03e.png">
<img width="1314" alt="image" src="https://user-images.githubusercontent.com/34237589/218953346-8dd5918b-f7d7-4e21-a63d-a405aa3dcff1.png">

**Comment by Rishab49** [comment]
I get your point and acknowledge my mistake of not understanding the problem statement.

**Comment by gricn** [comment]
Well, I have found that in the `.next/server/` folder, the `app-paths-manifest.json` parse the index as path `/path`. 
```
// .next/server/app-paths-manifest.json
{
  "/page": "app/page.js"
}
````

But in fact, it should be parsed as `"/":"app/page.js"`. And when I try to edit the `page.tsx` to `layout.tsx` in the `/pages` folder, however, the dev mode got 404 but production mode works fine.

I also found that in the `next/dist/server/get-app-route-from-entrypoint.js`, it will parse the page path of `app/page.js` to `/page` and this path is the same as the result of parsing from `pages/page.tsx`. 

And `get-app-route-from-entrypoint.js` file will be used by webpack plugins such as  `pages-manifest-plugin.js`, and with something related to webpack's `compilation.entrypoints`. Because I didn't get in touch with webpack before, so I stuck in this step.

------

So, can anyone tell me about whether my analysis is correct? By the way, could anyone tell me how to debug the project in `Next.js` repository? I have put this project to the `example` folder and edit the `.vscode/launch.json` followed the tutorial in `contributing` folder. But it won't work, the debug mode exit without any useful hints. Thanks.

**Comment by JanKaifer** [comment]
@gricn Thanks for the tips. I'm looking into this now.

> By the way, could anyone tell me how to debug the project in Next.js repository?
You should be able to use `pnpm next /examples/your-example`, which contribution doc mentions `.vscode/launch.json` haven't seen it there?

**Comment by JanKaifer** [comment]
Also that value of manifest is expected, keys in manifest have suffixes based on file type, like `/page`, `/layout` or `/loading`

**Comment by gricn** [comment]
> You should be able to use pnpm next /examples/your-example, which contribution doc mentions .vscode/launch.json haven't seen it there?

I have seen that, but I couldn't run this project even if I put it in `examples` folder before. After running the `git-clean` script, it works now.

**Comment by huozhi** [comment]
It’s fixed in v13.2.4-canary.6
